site/_posts/2018-07-06-ray-0.5-release.markdown

```diff
@@ -0,0 +1,79 @@
+---
+layout: post
+title: "Ray: 0.5 Release"
+excerpt: "This post announces the release of Ray 0.5."
+date: 2018-07-06 14:00:00
+---
+
+We are pleased to announce the 0.5 release of [Ray][1]. To upgrade to the latest
+version, run
+
+```
+pip install -U ray
+```
+
+## Refactoring
+
+One primary development is a substantial refactoring of the backend, which is
+currently off by default, but can be turned on by starting Ray with the
+environment variable `RAY_USE_XRAY=1`. This will be turned on soon after more
+testing and development and will be on by default in the next release. Some
+highlights of the refactor include the following:
+- Merging the "local scheduler" and the "plasma manager" into a single "raylet"
+  process (consisting of a "node manager" and an "object manager").
+- Faster object transfers by using multiple threads to send objects in chunks.
+- The introduction of a "lineage cache" to improve latency by removing the
+  global control store from the scheduling critical path.
+- The removal of the global scheduler (which is replaced by direct node-manager
+  to node-manager communication)
+- Providing a standardized table-based interface to the Redis-backed global
+  control store.
+- A preliminary approach for automatically flushing global control store
+  metadata to avoid out-of-memory errors for long-running jobs.
+Note that the refactoring is currently under development and does not yet
+support fault tolerance.
+
+## RLlib Improvements
+
+A number of improvements have been made to [RLlib][2]:
+- Added DDPG [[1877]](https://github.com/ray-project/ray/commit/c9a7744e5269a035eafceba35c7df3f52607b825).
+- Added a policy server example [[2232]](https://github.com/ray-project/ray/commit/e5724a9cfefd5c11467ff7975397ddfb5d2fe0e7).
+- Multi-agent support [[2286]](https://github.com/ray-project/ray/commit/a9a26b756098dcbd108966c52e3e7e9c9bfaf7ee).
```

Comment:
I'd also mention vectorized execution support, people have asked about that before: https://github.com/ray-project/ray/pull/2170

Reply:
done